src/ParametricOptInterface.jl

```diff
@@ -70,6 +72,191 @@ mutable struct ParametricOptimizer{T, OT <: MOI.ModelLike} <: MOI.AbstractOptimi
     end
 end
 
+function MOI.is_empty(model::ParametricOptimizer)
+    return MOI.is_empty(model.optimizer) &&
+    isempty(model.parameters) &&
+    isempty(model.parameters_name) &&
+    isempty(model.variables) &&
+    isempty(model.updated_parameters) &&
+    isempty(model.variables) &&
+    model.last_index_added == 0 &&
+    isempty(model.affine_constraint_cache) &&
+    isempty(model.quadratic_constraint_cache_pv) &&
+    isempty(model.quadratic_constraint_cache_pp) &&
+    isempty(model.quadratic_constraint_cache_pc) &&
+    isempty(model.quadratic_constraint_variables_associated_to_parameters_cache) &&
+    isempty(model.quadratic_added_cache) &&
+    model.last_quad_add_added == 0 &&
+    isempty(model.affine_objective_cache) &&
+    isempty(model.quadratic_objective_cache_pv) &&
+    isempty(model.quadratic_objective_cache_pp) &&
+    isempty(model.quadratic_objective_cache_pc) &&
+    isempty(model.quadratic_objective_variables_associated_to_parameters_cache)
+end
+
+function MOI.supports_constraint(
+    model::ParametricOptimizer,
+    F::Type{<:MOI.AbstractFunction},
```

Comment:
Replace `MOI.AbstractFunction` with `Union{MOI.SingleVariable, ...}`. Here, you say you support custom functions defined by MOI extensions but it's not true.

Reply:
You also don't support `VectorQuadraticFunction`